packages/plugins/auto-doc/src/stores/doc-templates.store.ts

```diff
@@ -0,0 +1,216 @@
+}
+
+function addDocumentTemplate(description: string): string | null {
+    let generatedId: string | null = null;
+    const xmlDoc = get(xmlDocument);
+    if (!xmlDoc) {
+        throw new Error("XML Document is not defined");
+    }
+
+    const currentPrivateArea = get(autoDocArea);
+    if (currentPrivateArea) {
+        const id = uuidv4();
+
+        const newDocDef = createElement(xmlDoc, 'DocumentTemplate', {
+            id: id,
+            date: new Date().toISOString(),
+            description: description
+        });
+        currentPrivateArea.appendChild(newDocDef);
+        eventStore.createAndDispatchActionEvent(currentPrivateArea, newDocDef);
+        generatedId = id;
+    }
+
+    return generatedId;
+}
+
+function editDocumentTemplateDescription(docTemplateId: string, newDescription: string) {
+    const docTemplate = getDocumentTemplate(docTemplateId);
+    if (docTemplate) {
+        docTemplate.setAttribute('description', newDescription);
+        eventStore.updateAndDispatchActionEvent(docTemplate, { description: newDescription });
+    }
+}
+
+function addBlockToDocumentTemplate(docTemplate: Element, content: string, type: string, position: number) {
+    const generatedId = uuidv4();
+    const xmlDoc = get(xmlDocument);
+    if (!xmlDoc) {
+        throw new Error("XML Document is not defined");
+    }
+    const blockElement = createElement(xmlDoc, 'Block', { 
+        id: generatedId,
+        type: type 
+    });
+    blockElement.textContent = content;
+    docTemplate.appendChild(blockElement);
+
+    insertBlockAtPosition(docTemplate, blockElement, position);
+    eventStore.createAndDispatchActionEvent(docTemplate, blockElement);
+    eventStore.moveAndDispatchActionEvent(docTemplate, docTemplate, blockElement, position);
+
+    return generatedId;
+}
+
+function editBlockContentOfDocumentTemplate(docTemplate: Element, blockId: string, content: string) {
+    const blockElement = docTemplate.querySelector(`Block[id="${blockId}"]`);
+    if (blockElement) {
+        blockElement.textContent = content;
+        eventStore.updateAndDispatchActionEvent(blockElement, { content: content });
+    }
+}
+
+function moveBlockInDocumentTemplate(docTemplate: Element, blockId: string, position: number) {
+    const blockIdElement = docTemplate.querySelector(`Block[id="${blockId}"]`);
+    if (blockIdElement) {
+        insertBlockAtPosition(docTemplate, blockIdElement, position);
+        eventStore.moveAndDispatchActionEvent(docTemplate, docTemplate, blockIdElement, position);
+    }
+}
+
+function deleteBlockFromDocumentTemplate(docTemplate: Element, blockId: string) {
+    const blockElement = docTemplate.querySelector(`Block[id="${blockId}"]`);
+    if (blockElement && blockElement.parentNode === docTemplate) {
+        eventStore.deleteAndDispatchActionEvent(docTemplate, blockElement);
+        docTemplate.removeChild(blockElement);
+    }
+}
+
+function deleteDocumentTemplate(docTemplateId: string) {
```

Comment:
One thing that comes to mind right now is if we might need a "delete multiple document template" Function in the future. I think it's something we can discuss  if we want to allow it or not

Reply:
I would suggest that i add the function as soon as we add the feature.
Otherwise it could end in dead code

Reply:
Agree